async_fuse/src/memfs/node.rs

```diff
@@ -434,13 +438,30 @@ impl Node {
     }
 
     /// Check whether to load file content data or not
-    pub fn need_load_file_data(&self) -> bool {
+    pub fn need_load_file_data(&self, offset: usize, len: usize) -> bool {
         debug_assert_eq!(
             self.attr.kind,
             SFlag::S_IFREG,
             "fobidden to check non-file node need load data or not",
         );
-        self.need_load_node_data_helper()
+
+        if offset > self.attr.size.cast() {
+            return false;
+        }
+
+        match self.data {
+            NodeData::RegFile(ref cache) => {
+                cache
+                    .get_file_cache(self.name.to_str().unwrap(), offset, len)
```

Comment:
Unix file name are not always UTF-8 strings.
A GBK file name may cause fatal crashs without messages.

Reply:
Yes, that's true, but it almost impossible to know the encoding, actually OS treat it as a sequence of bytes. After all it's a good point, a possible improvement is to adjust the key of the cache to a byte array.

Reply:
Another choice is forbidding non-UTF8 file names. I'm not sure which is better.

Reply:
It's defined by the users not even OS. OS only accepts bytes, so sticking to the bytes is a better choise from my view.